exciting_environments/core_env.py

```diff
@@ -149,109 +143,101 @@ def _test_rew_func(self, func):
             compatible(bool): Environment compatibility.
         """
         try:
-            out = func(
-                jnp.zeros([self.batch_size, int(len(self.obs_description))]))
+            out = jax.vmap(func, in_axes=(0, 0, self.in_axes_env_properties.action_constraints))(
+                jnp.zeros((self.batch_size, int(len(self.obs_description)))),
+                jnp.zeros((self.batch_size, int(
+                    len(list(vars(self.env_properties.action_constraints).values()))))),
+                self.env_properties.action_constraints
+            )
         except:
             raise Exception(
-                "Reward function should be using obs matrix as only parameter")
+                "Reward function should be using observation matrix, action and self.env_properties.action_constraints as parameters")
         try:
             if out.shape != (self.batch_size, 1):
                 raise Exception(
-                    "Reward function should be returning vector in shape (batch_size,1)")
+                    "Reward function should be returning vector in shape (1,)")
         except:
             raise Exception(
-                "Reward function should be returning vector in shape (batch_size,1)")
+                "Reward function should be returning vector in shape (1,)")
         return True
 
     @partial(jax.jit, static_argnums=0)
-    def step(self, action_norm, states):
-        """Addtional function in step execution to enable JAX jit.
+    def step(self, states, action, env_properties):
+        """Computes one simulation step for one batch.
 
         Args:
-            states(ndarray(float)): State Matrix (shape=(batch_size,states)).
-            action_norm(ndarray(float)): Action Matrix (shape=(batch_size,actions)).
-
+            states: The states from which to calculate states for the next step.
+            action: The action to apply to the environment.
+            env_properties: Contains action/state constraints and static parameter.
 
         Returns:
             Multiple Outputs:
 
-            observation(ndarray(float)): Observation/State Matrix (shape=(batch_size,states)).
-
-            reward(ndarray(float)): Amount of reward received for the last step (shape=(batch_size,1)).
+            observation: The gathered observation.
 
-            terminated(bool): Flag, indicating if Agent has reached the terminal state.
+            reward: Amount of reward received for the last step.
 
-            truncated(ndarray(bool)): Flag, indicating if state has gone out of bounds (shape=(batch_size,states)).
+            terminated: Flag, indicating if Agent has reached the terminal state.
 
-            {}: An empty dictionary for consistency with the OpenAi Gym interface.
+            truncated: Flag, indicating if state has gone out of bounds.
```

Comment:
Going out of bounds is not the only case where an environment could be truncated. A different case would be an artifical timer that is independent of the actual task.

See [here](https://gymnasium.farama.org/tutorials/gymnasium_basics/handling_time_limits/#truncation) for more information.